nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/validators/OIDValidator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.snmp.validators;
+
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.snmp.configuration.SecurityConfiguration;
+import org.apache.nifi.util.StringUtils;
+import org.snmp4j.security.SecurityLevel;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class OIDValidator {
+
+    private static final String SNMP_V3 = "SNMPv3";
```

Comment:
The SNMP version strings appear to be used in several places so it could be useful to define a reusable enum.